memcache_proxy/dtog.go

```diff
@@ -371,6 +379,16 @@ func unpackFirstStorageCmdArgs(args ...[]byte) (key []byte, flags uint32,
 		return
 	}
 	bytes = uint32(bytesUpc)
+	// Make sure we only accept up to 1000000 bytes to read (max value size
+	// for App Engine cache). Other parts of the code will allocate memory
+	// based on this number and this will only fail once we actually hit
+	// App Engine Memcache.
+	if bytes > maxMemcacheValueSize {
+		err = clientError{"bad command line format",
+			fmt.Sprintf("Value size should not be greater than %d, but got %d.",
+				maxMemcacheValueSize, bytes)}
+		return
```

Comment:
err is an output param and is being assigned here, so we're good. The other output params don't need to be set.